### DISCUSSION TITLE
Using a different css file per domain

### DISCUSSION THREAD

**Discussion by ronenteva** [nuxt/nuxt#15987] [Questions]
Any suggestions how to use a different css file on the same project, based on the domain being used?

**Comment by zumm** [comment]
You can create [server route](https://nuxt.com/docs/guide/directory-structure/server) for css file that will return different data depends on current host. 

```ts
// /server/routes/dynamic.css.ts
import { defineEventHandler, getHeader } from 'h3'

export default defineEventHandler(async (event) => {
  const host = getHeader(event, 'host')

  if (host === 'domain1') return 'css data 1'
  else return 'css data 2'
})
```

Next you need to add this css link by `nuxt.config.ts` and `app.head` for whole project or by `useHead` for some pages.

```ts
// /nuxt.config.ts
import { defineNuxtConfig } from 'nuxt/config'

export default defineNuxtConfig({
  app: {
    head: {
      link: [{
        rel: 'stylesheet',
        href: '/dynamic.css'
      }]
    }
  }
})
```

```ts
// /pages/some-page.vue
useHead({
  link: [{
    rel: 'stylesheet',
    href: '/dynamic.css'
  }]
})
```

You also may need to store css data in different css files and use [server assets](https://nitro.unjs.io/guide/introduction/assets) to have access to them on server side. But i have no idea how to force nuxt to process these files with bundler for minification and other css manipulations.

P.S.: Don't forget about caching.

**Reply by ronenteva** [reply]
Thanks!
I'll definitely consider that but I need the files to go through the bundler.

Another idea is to use [layers ](https://nuxt.com/docs/guide/going-further/layers) and extend one base project into a few projects, one for each domain. Think that would work?

**Reply by zumm** [reply]
I have never used layers, so idk. But mb preprocessing these files with bundler is the way. I mean you can perform minifying and other transformations without nuxt itself.
